### ISSUE TITLE
Slow loading

### ISSUE THREAD

**Issue by punitagoel2403** [google/ExoPlayer#7989]
The exoplayer is loading the video from server. The video file is 5 MB max, but loading is slow. Can the team please test if the speed issue is with server or I am doing anything wrong with the implementation. The implementation is very basic though. 

I can provide the url to the video, and the video separately as well. But not publicly. Please guide and help me.

**Comment by ojw28** [comment]
At the point you're filing an issue here, you should be pretty confident that it's an ExoPlayer issue. This doesn't seem to be the case currently. Note that there are multiple other ways that you can test whether the speed issue is with your server, for example by trying to play the same file with something other than ExoPlayer. If loading is also slow in that case, then it's likely to be an issue with your server or content.

Please could you do some further debugging on your side first, and follow up here if you have good evidence to suggest it's an ExoPlayer issue. Thanks.

**Comment by punitagoel2403** [comment]
It is different at different times. Sometimes it takes 4-5 sec, 8-9 sec other times, and some times just 1-2 sec to start. ( For 1.29 MB video), no caching done, I am testing for first time played video.
I tested with Exoplayer demo app too, it's the same, time varies.
Also, I cannot test with anything other than exoplayer, since we switched to exoplayer only to speed up the loading, which was slow in other cases.
It improved a bit by reducing the bufferforplayebackms to 1024. But the start time still varies, with good internet speed.

**Comment by AquilesCanta** [comment]
> Also, I cannot test with anything other than exoplayer, since we switched to exoplayer only to speed up the loading, which was slow in other cases.

This seems to suggest the problem you are observing is indeed a server side issue.

There are many ways you can test this:
- Use an external tool for measuring the time it takes for you to download the file. Like curl. There's [plenty of info](https://stackoverflow.com/questions/18215389/how-do-i-measure-request-and-response-times-at-once-using-curl) on how to do this.
- Add some log lines in your application to see how much time ExoPlayer spends waiting for the network transfers. For this you should look into the `DataSource` you are using.
- Finally, if you control the server, then you can surely do some research on that side. Perhaps the server is receiving high traffic, which could explain the delays.

The bottom line is we cannot spend time debugging this for you. There are two many ExoPlayer users for us to investigate issues which are not related to ExoPlayer. Please collect some evidence that this is an ExoPlayer issue, and then reach back to us. When you file a new issue, please don't ignore the issue template that appears in the "new issue" form.

**Comment by punitagoel2403** [comment]
okay..thank you..
